packages/pluggableWidgets/line-chart-native/src/utils/SeriesDataLoadingFunction.ts

```diff
@@ -0,0 +1,203 @@
+            typeof groupByAttributeValue.value !== "boolean"
+        ) {
+            return null;
+        }
+
+        const group = dataSourceItemGroupsResult.find(group => {
+            if (groupByAttributeValue.value instanceof Date && group.groupByAttributeValue instanceof Date) {
+                return group.groupByAttributeValue.getTime() === groupByAttributeValue.value.getTime();
+            } else if (groupByAttributeValue.value instanceof Big && group.groupByAttributeValue instanceof Big) {
+                return group.groupByAttributeValue.eq(groupByAttributeValue.value);
+            }
+            return group.groupByAttributeValue === groupByAttributeValue.value;
+        });
+
+        if (group) {
+            group.items.push(item);
+        } else {
+            const newDataSourceItemGroup: DataSourceItemGroup = {
+                groupByAttributeValue: groupByAttributeValue.value,
+                items: [item]
+            };
+
+            if (dynamicSeriesName) {
+                const dynamicSeriesNameValue = dynamicSeriesName(item);
+
+                if (!dynamicSeriesNameValue.value && typeof dynamicSeriesNameValue.value !== "string") {
+                    return null;
+                }
+
+                newDataSourceItemGroup.dynamicSeriesNameValue = dynamicSeriesNameValue.value;
+            }
+
+            if (dynamicStylePropertyName) {
+                const dynamicStylePropertyNameValue = dynamicStylePropertyName(item);
+
+                if (!dynamicStylePropertyNameValue.value && typeof dynamicStylePropertyNameValue.value !== "string") {
+                    return null;
+                }
+
+                newDataSourceItemGroup.dynamicStylePropertyNameValue = dynamicStylePropertyNameValue.value;
+            }
+
+            dataSourceItemGroupsResult.push(newDataSourceItemGroup);
+        }
+    }
+
+    return dataSourceItemGroupsResult;
+}
+
+function extractDataPoints(series: SeriesType, dataSourceItems?: ObjectItem[]): LineChartDataPoints | null {
+    const xValue = series.type === "static" ? ensure(series.staticXValue) : ensure(series.dynamicXValue);
+    const yValue = series.type === "static" ? ensure(series.staticYValue) : ensure(series.dynamicYValue);
+
+    if (!dataSourceItems) {
+        const dataSource =
+            series.type === "static" ? ensure(series.staticDataSource) : ensure(series.dynamicDataSource);
+
+        if (!dataSource.items) {
+            return null;
+        }
+
+        dataSourceItems = dataSource.items;
+    }
+
+    const dataPoints: LineChartDataPoints = [];
+
+    for (const item of dataSourceItems) {
+        const x = xValue(item);
+        const y = yValue(item);
+
+        if (!x.value || !y.value) {
+            return null;
+        }
+
+        dataPoints.push({
+            x: x.value instanceof Date ? x.value : (Number(x.value.toFixed()) as any), // Cast as any because data type will never differ for data points within a series
```

Comment:
`toString()` instead `toFixed()`?

Reply:
Changed it. I thought `Number()` couldn't handle exponential notation, but it does.